compiler/damlc/lib/DA/Cli/Damlc/Test/TestResults.hs

```diff
@@ -546,21 +554,32 @@ printTestCoverage showCoverage testResults@TestResults { templates, interfaceIns
     , externalImplementationChoicesReport
     ]
 
-    where
+printTemplateChoiceIdentifier :: (T.Text, TemplateIdentifier) -> String
+printTemplateChoiceIdentifier (choice, templateId) =
+    printTemplateIdentifier templateId <> ":" <> T.unpack choice
 
-    printTemplateChoiceIdentifier :: (T.Text, TemplateIdentifier) -> a -> String
-    printTemplateChoiceIdentifier (choice, templateId) _ =
-        printTemplateIdentifier templateId () <> ":" <> T.unpack choice
-
-    printInstanceChoiceIdentifier :: (T.Text, InterfaceInstanceIdentifier) -> a -> String
-    printInstanceChoiceIdentifier (choice, InterfaceInstanceIdentifier { instanceTemplate }) _ =
-        printTemplateIdentifier instanceTemplate () <> ":" <> T.unpack choice
-
-    printTemplateIdentifier :: TemplateIdentifier -> a -> String
-    printTemplateIdentifier TemplateIdentifier { package, qualifiedName } _ =
-        let prefix =
-                case package of
-                  LocalPackageId -> ""
-                  ExternalPackageId { name } -> name <> ":"
-        in
-        T.unpack $ prefix <> qualifiedName
+printInstanceChoiceIdentifier :: (T.Text, InterfaceInstanceIdentifier) -> String
+printInstanceChoiceIdentifier (choice, InterfaceInstanceIdentifier { instanceTemplate }) =
+    printTemplateIdentifier instanceTemplate <> ":" <> T.unpack choice
+
+printTemplateIdentifier :: TemplateIdentifier -> String
+printTemplateIdentifier TemplateIdentifier { package, qualifiedName } =
+    let prefix =
+            case package of
+              LocalPackageId -> ""
+              ExternalPackageId { name } -> name <> ":"
+    in
+    T.unpack $ prefix <> qualifiedName
+
+skipMatchingChoices :: (String -> Bool) -> TestResults -> TestResults
+skipMatchingChoices namePred TestResults{..} =
+  TestResults
+    { templates = M.mapWithKey (S.filter . shouldntSkip) templates
+    , interfaces = M.mapWithKey (S.filter . shouldntSkip . unInterfaceIdentifier) interfaces
+    , interfaceInstances
+    , created
+    , exercised = M.mapWithKey (\choiceName -> M.filterWithKey (\tid _ -> shouldntSkip tid choiceName)) exercised
+    }
+  where
+    shouldntSkip :: TemplateIdentifier -> T.Text -> Bool
```

Comment:
Suggestion: rename to `shouldKeep` or `shouldStay` to avoid the double negation.

Reply:
Fixed :)